verl/utils/tracking.py

```diff
@@ -255,3 +260,60 @@ def log_generations_to_mlflow(self, samples, step):
                 mlflow.log_artifact(validation_gen_step_file)
         except Exception as e:
             print(f"WARNING: save validation generation file to mlflow failed with error {e}")
+
+    def log_generations_to_database(self, samples, step, epoch, data_dir):
+        """Log the rollout samples to a parquet file at data_dir"""
+        import os
+        from datasets import Dataset
+
+        num_samples = len(samples)
+
+        # Create a Dataset from the samples
+        data = {
+            "input": [sample[0] for sample in samples],
+            "output": [sample[1] for sample in samples],
+            "score": [sample[2] for sample in samples],
+            "epoch": [epoch] * num_samples,
+            "step": [step] * num_samples
+        }
+        dataset = Dataset.from_dict(data)
+
+        # Define the path for the parquet file
+        file_path = os.path.join(data_dir, f"validation_step_{step}.parquet")
+
+        # Write the Dataset to a parquet file
+        dataset.to_parquet(file_path)
+
+
+@dataclasses.dataclass
+class RolloutLogger:
+
+    def log(self, loggers, samples, step, epoch=None, data_dir=None):
+        if 'database' in loggers:
+            if data_dir is None:
+                raise ValueError("Data directory must be provided for database logging")
+            if epoch is None:
+                raise ValueError("Epoch number be provided for database logging")
+            self.log_generations_to_database(samples, step, epoch, data_dir)
+
+    def log_generations_to_database(self, samples, step, epoch, data_dir):
+        """Log the rollout samples to a parquet file at data_dir"""
```

Comment:
in the docstring please specify the expected input format for samples. maybe as a logger it should just take a dict as the input, and let user prepare the `data` dict outside of the logger? i'm not sure how logging requirements will evolve overtime, since there's on going efforts for vlm, browser-use, etc